oxenss/snode/service_node.cpp

```diff
@@ -618,6 +618,50 @@ void ServiceNode::on_swarm_update(block_update&& bu) {
     // initiate_peer_test();
 }
 
+static void blocking_request_for_recent_block_hashes(
+        uint64_t height, server::OMQ& omq, std::map<uint64_t, std::string>& height_to_hashes) {
+
+    // Setup synchronisation primitives
+    std::mutex mutex;            // Mutex for blocking until all requests are done
+    std::mutex map_mutex;        // Sync writes to map
+    std::condition_variable cv;  // Sleep until all writes are complete
```

Comment:
A `std::promise<void>` would fit the purpose here better than a condition_variable that is only fired once, because then you can drop `mutex` entirely and the 5 lines after "block until all requests are returned" becomes a single line `prom.get_future().wait();`

Reply:
This was referring to an individual commit, since outdated.